### ISSUE TITLE
PHP 8.3 support

### ISSUE THREAD

**Issue by CViniciusSDias** [php-decimal/ext-decimal#72]
Code:
```
<?php
use Decimal\Decimal;

$decimal = [new Decimal('0.1'), new Decimal('0.2')];
echo array_sum($decimal) . PHP_EOL;
echo array_reduce($decimal, fn($carry, $value) => $carry + $value, new Decimal('0')) . PHP_EOL;
```

Expected output:
```
0.3
0.3
```

Actual output:
```
Warning: array_sum(): Addition is not supported on type Decimal\Decimal in /app/index.php on line 8

Warning: array_sum(): Addition is not supported on type Decimal\Decimal in /app/index.php on line 8
0

Fatal error: Uncaught TypeError: Unsupported operand types: Decimal\Decimal + Decimal\Decimal
```

**Comment by Jan-E** [comment]
There were some changes in array_sum:
https://github.com/php/php-src/blob/d9e2da342afa1c1e66b09a94b2448f03f653b42b/UPGRADING#L197-L200
```
  . array_sum() and array_product() now warn when values in the array cannot be converted to int/float.
    Previously arrays and objects where ignored whilst every other value was cast to int.
    Moreover, objects that define a numeric cast (e.g. GMP) are now casted instead of ignored.
    RFC: https://wiki.php.net/rfc/saner-array-sum-product
```
But this error message surprises me.

**Comment by michnovka** [comment]
I do not consider this an error though. array_sum is meant to work with native int|float types. it used to cast Decimal to float in PHP < 8.3, which is really not ideal. If you need to sum array of decimals, then you will need custom functions.

This is not fixable in the extension, as it would need to change the implementation of array_sum method itself

**Comment by CViniciusSDias** [comment]
Isn't it possible to have `array_sum` to call the overloaded operator `+` of the `Decimal` class instead of converting it to `float`? That seems reasonable to me as a user.

**Comment by michnovka** [comment]
well this is not what core php code should do. The function array_sum returns int|float, so in the end it would again convert into float, which IS WRONG.
The solution is to make your own array_sum function which handles Decimal values and returns Decimal.

**Comment by CViniciusSDias** [comment]
It doesn't seem wrong to me if the conversion to float happens only in the final step, after all math, so precision wouldn't have been lost. And if someone needs the result as Decimal instead of float, it would just be a matter of not using those functions, but I, as a userland developer, would expect it to work.

But I understand and respect if you disagree. If that change is not going to be worked on, I think this issue can be closed.

**Comment by rtheunissen** [comment]
I think it's a reasonable idea. The array sum function could pass through a mixed value, whatever the operator returned, which would return a decimal instance. It's basically a generic reduce function.

**Comment by rtheunissen** [comment]
Released as part of 1.5.0
